src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,131 @@
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private static User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        user = new User();
+    }
+
+    @BeforeEach
+    void setUp() {
+        Storage.people.clear();
+        user.setLogin("okname");
+        user.setPassword("okpassword");
+        user.setAge(20);
+    }
+
+    @Test
+    void register_nullUser_NotOk() {
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(null);
+        });
+    }
+
+    @Test
+    void register_nullLogin_NotOk() {
+        user.setLogin(null);
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_emptyLogin_NotOk() {
+        user.setLogin("");
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_onlyWhitespacesLogin_notOk() {
+        user.setLogin("     ");
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_nullAge_NotOk() {
+        user.setAge(null);
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_invalidAge_NotOk() {
+        user.setAge(0);
+        user.setAge(-1);
+        user.setAge(1);
+        user.setAge(17);
+        user.setAge(Integer.MIN_VALUE);
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_validAge_Ok() {
+        user.setAge(18);
+        User expected = user;
+        User actual = registrationService.register(user);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void register_nullPassword_NotOk() {
+        user.setPassword(null);
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_lessThan5CharsPassword_NotOk() {
+        user.setPassword("12345");
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_onlyWhitespacesPassword_notOk() {
+        user.setPassword("      ");
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_alreadyContainingUser_notOk() {
+        StorageDao storageDao = new StorageDaoImpl();
```

Comment:
remove instances of `StorageDao` from this test class, you don't need it, you have to test `RegistrationService` without this class